ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/ModelConformanceValidator.scala

```diff
@@ -154,13 +154,7 @@ private[transaction] class ModelConformanceValidator(engine: Engine, metrics: Me
           case DamlStateValue.ValueCase.ARCHIVE =>
             // NOTE(JM): Engine only looks up packages once, compiles and caches,
             // provided that the engine instance is persisted.
-            try {
-              Some(archive.Decode.decodeArchive(value.getArchive)._2)
-            } catch {
-              case err: archive.Error =>
-                logger.warn("Decoding the archive failed.")
-                throw Err.DecodeError("Archive", err.getMessage)
-            }
+            Some(archive.Decode.assertDecodeArchive(value.getArchive)._2)
```

Comment:
Is there an advantage of not rewrapping the exception related to the decoding error here (besides less code)?

Reply:
I just refactored a bit too much.
I fixed that in c4e7f755702ff5f096b413e838744ed7056c462d